services/director-v2/src/simcore_service_director_v2/models/schemas/dynamic_services/scheduler.py

```diff
@@ -335,7 +336,7 @@ class SchedulerData(CommonServiceDetails, DynamicSidecarServiceLabels):
         description="Name of the current dynamic-sidecar being observed",
     )
     run_id: RunID = Field(
-        default_factory=uuid4,
+        default_factory=lambda: f"{arrow.utcnow().int_timestamp}",
```

Comment:
I suggest to use a free-function instead of a lambda for the factory and reuse it everywhere else.
Note how most of this PR is covered with changes of uuid4 function you are replacing.
e.g. in the `run_id` fixtures of the tests.

Reply:
THOUGHT: I see what you aim but I am not very convinced that this is the right way to get that information. If you want to get more "useful info" why don't you simply associate it as metadata to the `uuid` you already had?

An id should have **only one responsibility** that is identify uniquely within the define context an entity. You want to give two with this change.

Moreover, a uuid4 is computed taking into account time and other parameters (hw?) to avoid collisions. You are changing that for basically a timestamp (i.e. time) therefore I think you are making more likely for it to collide, specially in a replicating system. What is going to happen when the dv2 is scaled horizontally (as we intend to do)?

Reply:
FOLLOW UP: what about a combination of `int-snapshot/short-uuid` for the StudyID?

Reply:
Replies:

# Q1
 ok

# Q2

Does not apply. See reasoning in MD's question.
Also if we scale director-v2 this part will get some distributed lock. It will not be a concern.


# Q3

We can go with that, I have nothing against it. Let's see if @matusdrobuliak66 likes the new format.

Reply:
please also check the questions from SAN and MD